### ISSUE TITLE
Post-roadmap: API stabilization and package hardening

### ISSUE THREAD

**Issue by moryev** [moryev/tcspc-lifetime-toolkit#2]
## Motivation

During roadmap-driven development, several implementation choices are intentionally optimized for scientific iteration and internal navigation rather than for a polished external API. This is useful while the toolkit is still evolving, but it should be revisited once the initial roadmap is complete.

Two concrete examples already identified:

1. **Roadmap-specific terminology in the public API**
   - Names such as `Day53ABReport`, `build_day53_ab_report()`, and docstrings referring to `Week 8` / `Day 53` are understandable within the development roadmap, but potentially confusing to external users.
   - These should eventually be renamed using scientific/domain concepts such as reference-vs-OOD evaluation, distribution shift, robustness, or generalization.
   - Roadmap provenance can remain in development notes, changelog/history, or documentation, but should not dominate the reusable public API.

2. **Hard-wired estimator sets in evaluation code**
   - Current robustness/evaluation code explicitly assumes the benchmark estimator set (`Ridge`, `RandomForestRegressor`, `HistGradientBoostingRegressor`, etc.).
   - This is acceptable for a frozen reproducible benchmark, but the generic evaluation layer should eventually support additional estimators without requiring edits to core evaluation functions.
   - Future users may want to add other scikit-learn estimators, gradient-boosting libraries, Gaussian processes, neural networks, 1-D CNNs, transformers, or custom estimators.

## Goal

After the scientific roadmap is complete, perform a systematic package-wide refactor focused on external usability, extensibility, and API stability while preserving reproducibility of the benchmark configurations developed during the roadmap.

## Proposed work

- [ ] Remove `Day XX` / `Week X` terminology from public class and function names.
- [ ] Replace roadmap-specific names with scientific/domain-oriented terminology.
- [ ] Keep roadmap provenance only where useful in documentation/history.
- [ ] Separate **generic evaluation infrastructure** from **frozen benchmark configurations**.
- [ ] Replace hard-coded estimator-name loops with estimator/model specifications, registries, or user-supplied estimator mappings.
- [ ] Allow custom sklearn-compatible estimators to participate without modifying evaluation internals.
- [ ] Design an extensibility layer that can later accommodate DL estimators whose training/prediction interface may differ from sklearn.
- [ ] Review which functions/classes should be public versus private.
- [ ] Standardize naming of result dataclasses, benchmark objects, summaries, diagnostics, and reports.
- [ ] Consolidate duplicated validation, prediction-table, summary-table, and comparison-table logic.
- [ ] Review module responsibilities and split modules that have become too broad during incremental development.
- [ ] Review type annotations and generic protocols/interfaces where appropriate.
- [ ] Review package-level exports and establish a clean high-level public API.
- [ ] Improve user-facing examples and documentation around extension points.
- [ ] Preserve frozen default benchmark definitions so published/reproducible experiments remain exactly reproducible.
- [ ] Add/adjust tests to verify that user-supplied estimators and representations can be evaluated without internal code changes.

## Architectural direction

Prefer a structure conceptually like:

```text
generic evaluation infrastructure
        ↑
        │
frozen benchmark configuration(s)
```

rather than embedding the frozen benchmark estimator list directly inside generic evaluation code.

A possible future abstraction might resemble:

```python
@dataclass(frozen=True)
class EstimatorSpec:
    name: str
    factory: Callable[[], Any]
    representations: tuple[str, ...]
```

with evaluation functions iterating over supplied specifications rather than knowing estimator identities themselves.

## Timing

**Do not perform this refactor prematurely.**

The current roadmap-oriented structure is useful while scientific functionality is still being discovered. Refactoring should happen after the initial roadmap is complete, when recurring abstractions and actual extension requirements are visible. This avoids repeatedly redesigning abstractions during active scientific development.

This issue is therefore intentional technical debt and a reminder for the post-roadmap stabilization phase, not an indication that the current implementation is incorrect.
